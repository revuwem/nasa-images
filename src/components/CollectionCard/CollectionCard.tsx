type Props = {
  data: {
    preview: string;
    title: string;
    location: string;
    author: string;
  };
};

const CollectionCard: React.FC<Props> = (props) => {
  const { preview, title, location, author } = props.data;
  return (
    <div className="group">
      <div className="h-72 flex mb-3 overflow-hidden">
        <img
          src={preview}
          alt=""
          className="w-full h-full object-cover transition-all duration-500 scale-110 group-hover:scale-100"
        />
      </div>
      <h3 className="text-xl text-grey-light font-bold mb-4 transition group-hover:text-blue-500">
        {title}
      </h3>
      {author && <p className="text-grey-dark mb-2">{author}</p>}
      {location && <p className="text-grey-dark italic">{location}</p>}
    </div>
  );
};

export default CollectionCard;
