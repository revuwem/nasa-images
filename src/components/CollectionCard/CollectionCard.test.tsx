import { describe, it } from "vitest";
import renderer from "react-test-renderer";
import CollectionCard from "@/components/CollectionCard";

const data = {
  preview: "preview.png",
  title: "Title",
  location: "location",
  author: "author",
};

describe("CollectionCard", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<CollectionCard data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
