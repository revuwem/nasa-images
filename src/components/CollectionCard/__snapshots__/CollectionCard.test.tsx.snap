// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CollectionCard > should render correctly 1`] = `
<div
  className="group"
>
  <div
    className="h-72 flex mb-3 overflow-hidden"
  >
    <img
      alt=""
      className="w-full h-full object-cover transition-all duration-500 scale-110 group-hover:scale-100"
      src="preview.png"
    />
  </div>
  <h3
    className="text-xl text-grey-light font-bold mb-4 transition group-hover:text-blue-500"
  >
    Title
  </h3>
  <p
    className="text-grey-dark mb-2"
  >
    author
  </p>
  <p
    className="text-grey-dark italic"
  >
    location
  </p>
</div>
`;
